src/watchdog.rs

```diff
@@ -0,0 +1,110 @@
+//! Watchdog peripherals
+
+use crate::{
+    hal::watchdog::{Watchdog, WatchdogEnable},
+    stm32::{DBGMCU, IWDG},
+    time::MilliSeconds,
+};
+
+/// Wraps the Independent Watchdog (IWDG) peripheral
+pub struct IndependentWatchdog {
+    iwdg: IWDG,
+}
+
+const LSI_KHZ: u32 = 32;
+const MAX_PR: u8 = 0b110;
+const MAX_RL: u16 = 0xFFF;
+const KR_ACCESS: u16 = 0x5555;
+const KR_RELOAD: u16 = 0xAAAA;
+const KR_START: u16 = 0xCCCC;
+
+impl IndependentWatchdog {
+    /// Creates a new `IndependentWatchDog` without starting it. Call `start` to start the watchdog.
+    /// See `WatchdogEnable` and `Watchdog` for more info.
+    pub fn new(iwdg: IWDG) -> Self {
+        IndependentWatchdog { iwdg }
+    }
+
+    /// Debug independent watchdog stopped when core is halted
+    pub fn stop_on_debug(&self, dbgmcu: &DBGMCU, stop: bool) {
+        #[cfg(any(feature = "stm32l4x1", feature = "stm32l4x2", feature = "stm32l4x3",))]
+        dbgmcu.apb1fzr1.modify(|_, w| w.dbg_iwdg_stop().bit(stop));
+        #[cfg(any(feature = "stm32l4x5", feature = "stm32l4x6"))]
+        dbgmcu.apb1_fzr1.modify(|_, w| w.dbg_iwdg_stop().bit(stop));
+    }
+
+    /// Sets the watchdog timer timout period. Max: 32768 ms
+    fn setup(&self, timeout_ms: u32) {
```

Comment:
I think there should be an assert here to limit the max timeout_ms value

Reply:
Good idea, will do.